api/src/test/java/org/openmrs/validator/ObsValidatorTest.java

```diff
@@ -513,4 +517,23 @@ public void validate_shouldPassValidationIfValueTextIsLessThanTheMaximumLength()
 		Assert.assertFalse(errors.hasFieldErrors("obsDatetime"));
 		Assert.assertFalse(errors.hasFieldErrors("valueText"));
 	}
+
+	@Test
+	public void validate_shouldFailValidationForDifferentObsPersonAndWithEncounterPatient() {
+		EncounterService es = Context.getEncounterService();
+		PersonService ps = Context.getPersonService();
+
+		Obs obs = new Obs(1);
+		obs.setPerson(ps.getPerson(2)); // Setting this person (2) as different from that of encounter person
+		obs.setEncounter(es.getEncounter( 3 ));
+		obs.getEncounter().setPatient(new Patient( 3 ) ); //Setting encounter person(3) as different from that of obs person
+
+		assertFalse( Objects.equals(obs.getPerson(),obs.getEncounter().getPatient()) ); //checking whether the two person objects are different
+		
+		Errors errors = new BindException(obs, "obs");
+		new ObsValidator().validate(obs, errors);
```

Comment:
can you please arrange the test like in https://github.com/openmrs/openmrs-core/blob/master/api/src/test/java/org/openmrs/validator/VisitValidatorTest.java#L545

using newlines to separate blocks of code that belong together. This is sometimes called "triple A" or AAA with 3 blocks of arrange, act, assert.

1. arrange: the test setup - for example you need a `Patient` that is < 16 years old
2. action: the method under test - for example a method that creates a `Visit` for the patient
3. assert: things you expect to happen after the action - for example you expect the method to throw an exception given this `Patient` since with an age < 16 years the method fails to create a `Visit` when the `Patient` has no relative like a family member connected to her/him

With this separation we can easily see the specific test case due to 1, what you are testing due to 2 and also what you expect to happen. Of course its sometimes necessary/helpful to use some newlines in 1 if the setup is a bit bigger but its good to have these general blocks.